sm/engine/dataset_manager.py

```diff
@@ -252,11 +254,21 @@ def _add_zoom_optical_images(self, ds, img_id, transform, zoom_levels):
         self._db.alter(DEL_OPTICAL_IMAGE, params=(ds.id,))
         self._db.insert(INS_OPTICAL_IMAGE, rows=rows)
 
+    def _add_thumbnail_optical_image(self, ds, img_id):
+        size = 200, 200
+        self._db.alter(UPD_DATASET_THUMB_OPTICAL_IMAGE, params=(None, ds.id,))
+        optical_img = self._img_store.get_image_by_id('fs', 'raw_optical_image', img_id)
+        optical_img.thumbnail(size, Image.ANTIALIAS)
+        buf = self._save_jpeg(optical_img)
+        img_thumb_id = self._img_store.post_image('fs', 'optical_image', buf)
+        self._db.alter(UPD_DATASET_THUMB_OPTICAL_IMAGE, params=(img_thumb_id, ds.id,))
+
     def add_optical_image(self, ds, img_id, transform, zoom_levels=[1, 2, 4, 8], **kwargs):
-        """ Generate scaled and transformed versions of the provided optical image """
+        """ Generate scaled and transformed versions of the provided optical image + creates the thumbnail """
         self.logger.info('Adding optical image to "%s" dataset', ds.id)
         self._add_raw_optical_image(ds, img_id, transform)
         self._add_zoom_optical_images(ds, img_id, transform, zoom_levels)
+        self._add_thumbnail_optical_image(ds, img_id)
```

Comment:
Not sure if this is intentional, but the thumbnail doesn't take into account the transform of the image.

Reply:
Yes, mine idea for that was to create a thumbnail from raw optical image (that is, the original size image) thereby not applying any transformation to it. 
I will try to generate the thumbnail from the transformed image as well, it can also give a good image

Reply:
@LachlanStuart